ubi_manifest/worker/tasks/depsolver/rpm_depsolver.py

```diff
@@ -272,6 +275,13 @@ def run(self) -> None:
             ft = self._executor.submit(self.get_source_pkgs, resolved, merged_blacklist)
             source_rpm_fts.append(ft)
 
+        # resolve file requirements
+        if self._required_files:
+            crit = create_or_criteria(
+                ["files"], [(filename,) for filename in self._required_files]
+            )
+            self.output_set.update(search_rpms(crit, pulp_repos).result())
```

Comment:
Have you tried this query on real data? I am a bit worried on performance of this query.

Also can you consider a bit smarter approach? In this place of code we already went through a lot of RPMs that may provide files we need so we don't have to query again. So it would be good be if we check during depsolving whether some of RPMs provides required files.

I think there is also a bug - we can get  RPMs for which we don't have resolved dependencies because we only do that in that big `while` loop.

I guess you may need to wire resolving of `files` into the `while` loop.

Reply:
> Have you tried this query on real data? I am a bit worried on performance of this query.

Still working on figuring out how to set this up, @rbikar

Reply:
@rbikar, what do we do if something like `Criteria.and_(Criteria.with_unit_type(RpmUnit),Criteria.with_field("files","jq-1.6.tar.gz"))` isn't performant?
It seems to take 30-40 seconds in my tests.

Reply:
As in most DBs, I think fields have to be indexed in order for searches to be fast over a large data set.

I believe the indexes on RPM units have been declared here: https://github.com/pulp/pulp_rpm/blob/637665cbf98361d65d37bf301fdbbb44cbd55ea8/plugins/pulp_rpm/plugins/db/models.py#L536

So those are probably the only fields where searches can be "fast" right now.
I expect it would be possible for us to add new indexes if needed, but I don't think we've ever done that before and I'm not suggesting it's definitely needed here.